src/segments/gitversion.go

```diff
@@ -0,0 +1,83 @@
+package segments
+
+import (
+	"encoding/json"
+	"errors"
+	"oh-my-posh/environment"
+	"oh-my-posh/properties"
+)
+
+type SystemCalls interface {
+	Getwd() error
+}
+
+type Gitversion struct {
+	props properties.Properties
+	env   environment.Environment
+
+	SemverInfo
+}
+
+type SemverInfo struct {
+	SemVer           string `json:"SemVer"`
+	FullSemVer       string `json:"FullSemVer"`
+	NuGetVersion     string `json:"NuGetVersion"`
+	NuGetVersionV2   string `json:"NuGetVersionV2"`
+	PreReleaseNumber string `json:"PreReleaseNumber"`
+	MajorMinorPatch  string `json:"MajorMinorPatch"`
+}
+
+func (n *Gitversion) Template() string {
+	return " {{ .SemVer }} "
+}
+
+func (n *Gitversion) Enabled() bool {
+	gitversion := "gitversion"
+	if !n.env.HasCommand(gitversion) {
+		return false
+	}
+	dir := n.env.Pwd()
+	result, err := getCacheValue(n, dir)
+
+	if result.FullSemVer == "" {
```

Comment:
we can do an early return here instead of indenting the full block.

```suggestion
	if len(result.FullSemVer) != 0 {
	    return err == nil
	}
```